site/public/js/calendar.js

```diff
@@ -28,15 +28,51 @@ function prevMonth(month, year) {
  * @param year : int the current year
  * @returns {number[]} : array<int> {next_month, year_of_next_month}
  */
-function nextMonth(month, year) {
+function nextMonth(month, year, day) {
     month = month + 1;
     if (month > 12) {
         month = month - 12;
         year = year + 1;
     }
-    return [month, year];
+    return [month, year, day, 'month'];
 }
 
+/**
+ * Gets the previous week of a given month
+ * @param month : int the current month (1 as January and 12 as December)
+ * @param year : int the current year
+ * @param day : int the current day
+ * @returns {number[]} : array<int> {previous_month, year_of_previous_month}
```

Comment:
The specified return type doesn't seem to match what you're actually returning here.  You're returning an array of three things.